easybuild/easyconfigs/q/QuantumESPRESSO/QuantumESPRESSO-7.3.1-foss-2023a.eb

```diff
@@ -0,0 +1,154 @@
+# https://gitlab.com/QEF/q-e/-/issues/666
+local_d3q_hash = "de4718351e7bbb9d1d12aad2b7ca232d06775b83"
+local_qe_gipaw_hash = "75b01b694c9ba4df55d294cacc27cf28591b2161"
+local_qmcpack_hash = "f72ab25fa4ea755c1b4b230ae8074b47d5509c70"
+local_w90_hash = "1d6b187374a2d50b509e5e79e2cab01a79ff7ce1"
+
+
+sources = [
+    {
+        "filename": "q-e-qe-%(version)s.tar.gz",
+        "extract_cmd": "mkdir -p %(builddir)s/qe-%(version)s && tar xzvf %s --strip-components=1 -C $_",
+        "source_urls": ["https://gitlab.com/QEF/q-e/-/archive/qe-%(version)s"],
+    },
+    {
+        "filename": "lapack-%s.tar.gz" % local_lapack_hash,
+        "git_config": {
+            "url": "https://github.com/Reference-LAPACK",
+            "repo_name": "lapack",
+            "commit": local_lapack_hash,
+        },
+    },
+    {
+        "filename": "mbd-%s.tar.gz" % local_mbd_hash,
+        "git_config": {
+            "url": "https://github.com/libmbd",
+            "repo_name": "libmbd",
+            "commit": local_mbd_hash,
+            'clone_into': 'mbd',
+        },
+    },
+    {
+        "filename": "devxlib-%s.tar.gz" % local_devxlib_hash,
+        "git_config": {
+            "url": "https://gitlab.com/max-centre/components",
+            "repo_name": "devicexlib",
+            "commit": local_devxlib_hash,
+            'clone_into': 'devxlib',
+        },
+    },
+    {
+        "filename": "d3q-%s.tar.gz" % local_d3q_hash,
+        "git_config": {
+            "url": "https://github.com/anharmonic",
+            "repo_name": "d3q",
+            "commit": local_d3q_hash,
+        },
+    },
+    {
+        "filename": "fox-%s.tar.gz" % local_fox_hash,
+        "git_config": {
+            "url": "https://github.com/pietrodelugas",
+            "repo_name": "fox",
+            "commit": local_fox_hash,
+        },
+    },
+    {
+        "filename": "qe-gipaw-%s.tar.gz" % local_qe_gipaw_hash,
+        "git_config": {
+            "url": "https://github.com/dceresoli",
+            "repo_name": "qe-gipaw",
+            "commit": local_qe_gipaw_hash,
+        },
+    },
+    {
+        "filename": "pw2qmcpack-%s.tar.gz" % local_qmcpack_hash,
+        "git_config": {
+            "url": "https://github.com/QMCPACK",
+            "repo_name": "pw2qmcpack",
+            "commit": local_qmcpack_hash,
+        },
+    },
+    {
+        "filename": "wannier90-%s.tar.gz" % local_w90_hash,
+        "git_config": {
+            "url": "https://github.com/wannier-developers",
+            "repo_name": "wannier90",
+            "commit": local_w90_hash,
+        },
+    },
+]
+checksums = [
+    {'q-e-qe-7.3.1.tar.gz': '2c58b8fadfe4177de5a8b69eba447db5e623420b070dea6fd26c1533b081d844'},
+    # Holding off checksum checks untill 5.0.x
+    # https://github.com/easybuilders/easybuild-framework/pull/4248
+
+    # {'lapack-%s.tar.gz' % local_lapack_hash: 'c05532ae0e5fe35f473206dda12970da5f2e2214620487d71837ddcf0ea6b21d'},
```

Comment:
@Crivella CI is failing because we should just be retaining the entries, but using a `None` value for the checksum values (same is true for all the rest as well)

Reply:
@ocaisa Is there a standard way to mix `dict` checksums with None?

Tried with 
```
[
    { XXX: YYY },
    { ZZZ: None},
]
```

which fails the typechecking
and than
```
[
    { XXX: YYY },
   None,
]
```

which ends up converting the checksum string into a list of single characters due to https://github.com/easybuilders/easybuild-framework/blob/65591dd32a78850144ce39238aa577df6fab3f9f/easybuild/framework/easyconfig/types.py#L528
(Not sure if this is a known bug or if i should open an issue)

If i try (to get around the previous bug)
```
[
    { XXX: [YYY] },
   None,
]
```

then i hit again a typechecking failure

Reply:
Until we re-add the checksums I have convert them into a list without dicts [0781213](https://github.com/easybuilders/easybuild-easyconfigs/pull/20138/commits/07812133646be2c42d59d8f7b045258cd4a1e11e), this should work even without fixing the previously mentioned behavior